airflow/contrib/executors/mesos_executor.py

```diff
@@ -118,10 +124,16 @@ def resourceOffers(self, driver, offers):
             driver.launchTasks(offer.id, tasks)
 
     def statusUpdate(self, driver, update):
-        logging.info("Task %s is in state %s, data %s", \
-            update.task_id.value, mesos_pb2.TaskState.Name(update.state), str(update.data))
-        
-        key = self.task_key_map[update.task_id.value]
+        logging.info("Task %s is in state %s, data %s",
+                     update.task_id.value, mesos_pb2.TaskState.Name(update.state), str(update.data))
+
+        try:
+            key = self.task_key_map[update.task_id.value]
+        except KeyError:
+            # The map may not contain an item if the framework re-registered after a failover.
+            # Discard these tasks.
+            logging.warn("Unrecognised task key %s" % update.task_id.value)
```

Comment:
Good point. For moving towards a production class executor, I guess we should also try to reschedule / handle these tasks. For example, it might make more sense to keep framework state in a local db (or as is more common amongst mesos frameworks, in zookeeper), instead of in-memory. Further, that will also be useful when we move towards HA for airflow scheduler (say https://github.com/airbnb/airflow/issues/392 ). May be we can open an issue and take this up sometime later.